girder/api/v1/assetstore.py

```diff
@@ -248,3 +250,96 @@ def getAssetstoreFiles(self, assetstore, limit, offset, sort, params):
+
+        collectionItemsInStore = list(self.model('item').find(query={
+            '_id': {
+                '$in': list(set([x['itemId'] for x in filesInStore]))
+            },
+            'baseParentType': 'collection'
+        }))
+        collectionItemsInStoreMap = dict((x['_id'], x) for x in collectionItemsInStore)
+
+        collectionUsingStore = list(self.model('collection').find(
+            query={
+                '_id': {
+                    '$in': list(set([x['baseParentId'] for x in collectionItemsInStore]))
+                }
+            }))
+        collectionUsingStoreMap = dict((x['_id'], x) for x in collectionUsingStore)
+
+        collectionFileMap = {}
+        for file in filesInStore:
+            if file['itemId'] in collectionItemsInStoreMap:
+                collectionId = collectionItemsInStoreMap[file['itemId']]['baseParentId']
+                if collectionId not in collectionFileMap:
+                    collectionFileMap[collectionId] = [file]
+                else:
+                    collectionFileMap[collectionId].append(file)
+
+        usersUsingStore = list(self.model('user').find(
+            query={
+                '_id': {
+                    '$in': list(set([x['creatorId'] for x in filesInStore]))
+                }
+            }))
+        userUsingStoreMap = dict((x['_id'], x) for x in usersUsingStore)
+        userFileMap = {}
+        for file in filesInStore:
+            userId = file['creatorId']
+            if userId not in userFileMap:
+                userFileMap[userId] = [file]
+            else:
+                userFileMap[userId].append(file)
+
+        # based on collected data stored in dictionary, aggregate size and format
+        # result. Ignore the fact there will be only one copy for the same file
+        # in the assetstore.
+        currentUser = self.getCurrentUser()
+        collectionModel = self.model('collection')
+        userModel = self.model('user')
+
+        def collectionMapper(collectionId, files):
+            collection = collectionModel.filter(collectionUsingStoreMap[collectionId], currentUser)
+            collection = {
+                '_id': collection['_id'],
+                'name': collection['name']
+            }
+            collection['spaceUsage'] = sum(file['size'] for file in files)
+            return collection
+        collections = [
+            collectionMapper(x, y) for x, y in collectionFileMap.items()]
+        collections.sort(key=lambda x: x['spaceUsage'], reverse=True)
+
+        def userMapper(userId, files):
+            if userId not in userUsingStoreMap:
+                return None
+            user = userModel.filter(userUsingStoreMap[userId], currentUser)
+            user = {
+                '_id': user['_id'],
+                'login': user['login'],
+                'firstName': user['firstName'],
+                'lastName': user['lastName'],
+                'email': user['email']
```

Comment:
We are already calling `filter` on the user document above -- if we are going to just pull out some select fields from the user document, there's no need to filter, it should be one or the other.